common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -363,6 +365,95 @@ static int parseRoutes(struct nlmsghdr *nlHdr, std::shared_ptr<route_info> rtInf
     return (0);
 }
 
+std::shared_ptr<std::unordered_map<std::string, std::string>>
+network_utils::get_arp_table(bool mac_as_key)
+{
+    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
+    auto arp_table     = *arp_table_ptr.get();
```

Comment:
Here you create a copy of the just created unordered_map.
It seems to be wrong as you assign to it in [436](https://github.com/prplfoundation/prplMesh/pull/1368/commits/02696b3df302a2b283868c4286ceb764c75cc591#diff-97a22c96f0d7fa3e45d0d435e03adc8dR436) and [438](https://github.com/prplfoundation/prplMesh/pull/1368/commits/02696b3df302a2b283868c4286ceb764c75cc591#diff-97a22c96f0d7fa3e45d0d435e03adc8dR438) but never assign back to the pointer.
I suggest either use the pointer only or:
```suggestion
    auto& arp_table     = *arp_table_ptr.get();
```
For more details, paste the following code into https://cppinsights.io:
```cpp
#include <memory>
#include <unordered_map>
#include <string>

int main()
{
    auto arp_table_ptr = std::make_shared<std::unordered_map<std::string, std::string>>();
    auto arp_table     = *arp_table_ptr.get();
  
    auto int_ptr = std::make_shared<int>();
    auto int_    = *int_ptr.get();
}
```

Reply:
Yes, I forgot the & sign when changed the return type.
Anyway fixed.